src/libopensc/pkcs15-pubkey.c

```diff
@@ -1509,8 +1484,20 @@ static struct ec_curve_info {
 		{"secp192k1",		"1.3.132.0.31", "06052B8104001F", 192},
 		{"secp256k1",		"1.3.132.0.10", "06052B8104000A", 256},
 
-		{"ed25519",		"1.3.6.1.4.1.11591.15.1", "06092B06010401DA470F01", 255},
-		{"curve25519",		"1.3.6.1.4.1.3029.1.5.1", "060A2B060104019755010501", 255},
+		/* OpenPGP extensions by Yubikey and GNUK are not defined in RFCs but we know the oid written to card */
+
+		{"edwards25519",	"1.3.6.1.4.1.11591.15.1", "06092B06010401DA470F01", 255},
+		{"curve25519",		"1.3.6.1.4.3029.1.5.1", "060a2B060104019755010501", 255},
```

Comment:
I think in this case you also miss the `1.` before the 3029. Please double-check. The other OID for ed25519 looks good on fast review.